selenium_test.py

```diff
@@ -0,0 +1,273 @@
+
+
+class Articles:
+    def __init__(self, title, description, category, keyword, link ):
+        self.title = title
+        self.description = description
+        self.category = category
+        self.keyword = keyword
+        self.link = link
+
+    def to_dict(self):
+        return {
+            'Title': self.title,
+            'Description': self.description,
+            'Category': self.category,
+            'Keyword': self.keyword,
+            'Link': self.link,
+
+        }
+
+
+
+
+
+def read_login_info():
+    with open(LOGIN_FILE) as text_file:
+        loginInfo = text_file.read().split(',')
+        return loginInfo
+
+
+def read_keywords():
+    with open(KEYWORDS_FILE) as fd:
+        return [keyword.strip() for keyword in fd.readlines()]
+
+
+def scrape_economist(keyword):
+    allResults = []
+
+    path = os.getcwd() + "/geckodriver"
+    driver = webdriver.Firefox(executable_path=path)
+    driver.get("http://www.economist.com/historicalarchive")
+
+    # tests to make sure we are on login page
+    assert "Economist" in driver.title
+
+    loginInfo = read_login_info()
+    email = loginInfo[0]
+    loginPassword = loginInfo[1]
+
+    # inputs username
+    username = driver.find_element_by_name("email")
+    username.clear()
+    username.send_keys(email)
+
+    # inputs passwords
+    password = driver.find_element_by_name("password")
+    password.clear()
+    password.send_keys(loginPassword)
+
+    # clicks log in button
+    logIn = driver.find_element_by_xpath("/html/body/form/div[2]/table/tbody/tr[5]/td[2]/a")
+    logIn.click()
+
+
+    searchForm(driver, keyword)
+
+
+    # calls collect results on search results
+    collectResults(driver, keyword, allResults)
+    enabledButton = driver.find_elements(By.XPATH, '//*[@title="Next"]')
+    hasNextPage = False
+    if len(enabledButton) > 0:
+        hasNextPage = True
+
+    while hasNextPage:
+        disabledButton = driver.find_elements(By.XPATH, "/html/body/div[4]/div[3]/form/div[3]/div/ul[2]/b")
+        enabledButton = driver.find_elements(By.XPATH, '//*[@title="Next"]')
+        print("Length of enabled button" + str(len(enabledButton)))
+        print("Length of diabled button" + str(len(disabledButton)))
+
+        if len(enabledButton) > 0:
+            enabledButton[0].click()
+            collectResults(driver, keyword, allResults)
+            hasNextPage = True
+        elif len(disabledButton) > 0:
+            collectResults(driver, keyword, allResults)
+            break
+
+    download(driver)
+
+
+    title_list = []
+    description_list = []
+    category_list = []
+    keyword_list = []
+    link_list = []
+
+    for obj in allResults:
+        title_list.append(obj.title)
+        description_list.append(obj.description)
+        category_list.append(obj.category)
+        keyword_list.append(obj.keyword)
+        link_list.append(obj.link)
+
+    groupResults(title_list, description_list, category_list, keyword_list, link_list)
```

Comment:
if you're going to turn this into a DataFrame in `groupResults` anyway, you might as well pass in the list and use `from_records` there.